app/app/static/custom/ourscript.js

```diff
@@ -114,6 +115,51 @@ const terminateSession = async(data) => {
 	}
 }
 
+function validateUserInput(data) {
+	// Check student name
+	if (data.studentName.length > 50) {
+		alert("'Student Name' field should is too long.");
+		return false;
+	}
+	for(let i = 0; i < data.studentName.length; i++) {
+		if (!isNaN(parseInt(data.studentName[i]))) {
+			alert("Please do not use numeric values in 'Student Name' field.");
+			return false;
+		}
+	}
+
+	// Check student number
+	if (data.studentNumber.length != 8) {
+		alert("'Student Number' field should be an 8 digit number.");
+		return false;
+	}
+	for(let i = 0; i < data.studentNumber.length; i++) {
+		if (isNaN(parseInt(data.studentNumber[i]))) {
+			alert("Please only numeric values in 'Student Number' field.");
+			return false;
+		}
+	}
+
+	// Check unit code
+	if (data.unitCode.length != 8) {
+		alert("'Unit Code' field should be 4 alphabetic characters followed by 4 numerics characters.");
+		return false;
+	}
+	for(let i = 0; i < 8; i++) {
+		if (i < 4 && !data.unitCode[i].match(/[a-zA-Z]/i)) {
+			alert("'Unit Code' field should be 4 alphabetic characters followed by 4 numerics characters.");
+			return false;
+		}
+		else if (i >= 4 && !data.unitCode[i].match(/[0-9]/i)) {
+			alert("'Unit Code' field should be 4 alphabetic characters followed by 4 numerics characters.");
+			return false;
+		}
+	}
```

Comment:
This too is better written in regex
![image](https://user-images.githubusercontent.com/44391389/135422779-29fa6198-e9c8-436e-bcc0-38408e773580.png)

see https://regexr.com/

Reply:
Ok cheers that does look better will edit all the sections that would be better with regex